src/main/java/support/Storage.java

```diff
@@ -0,0 +1,41 @@
+package support;
+
+import java.io.File;
+
+import java.io.FileWriter;
+
+import java.io.IOException;
+
+/**
+ * Represents a database.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Creates a directory data, and file duke.txt if it does not exist.
+     */
+    public Storage() {
+        try {
+            this.file = new File("./" + "data/duke.txt");
+            file.createNewFile();
```

Comment:
Perhaps this might break when you delete the data directory? 

Deleting the data directory (that came with the zip file) results in your code being unable to recover from this deletion. Hence, the tasks are no longer stored across sessions. You might be able to observe this if the .jar file is run in 2 separate sessions. 

Just pointing this out as this seems to be a marking criteria. I recommend using Files.createDirectories() and the relative path methods like System.getProperty() and Paths.get() as a way to get around this issue.